Switcher/TargetInformationBuilder.php

```diff
@@ -31,17 +33,22 @@ class TargetInformationBuilder
     private $allowedLocales;
 
     /**
-     * @param Request         $request           Request
-     * @param RouterInterface $router            Router
-     * @param array           $allowedLocales    Config Var
-     * @param bool            $showCurrentLocale Config Var
-     * @param bool            $useController     Config Var
+     * @param RequestStack $requestStack Request
+     * @param RouterInterface $router Router
+     * @param AllowedLocalesProvider $allowedLocalesProvider
+     * @param bool $showCurrentLocale Config Var
+     * @param bool $useController Config Var
      */
-    public function __construct(Request $request, RouterInterface $router, $allowedLocales = array(), $showCurrentLocale = false, $useController = false)
-    {
-        $this->request = $request;
+    public function __construct(
+        RequestStack $requestStack,
+        RouterInterface $router,
+        AllowedLocalesProvider $allowedLocalesProvider,
+        $showCurrentLocale = false,
+        $useController = false
+    ) {
+        $this->request = $requestStack->getCurrentRequest();
         $this->router = $router;
-        $this->allowedLocales = $allowedLocales;
+        $this->allowedLocales = $allowedLocalesProvider->getAllowedLocales();
```

Comment:
i think i'd keep the reference to the provider and get the allowed locales on the fly when we actually need them